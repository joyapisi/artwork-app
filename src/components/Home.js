import React, { useState, useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { BsArrowRightCircle } from 'react-icons/bs';
import { fetchArtworks } from '../redux/home/homeSlice';
import '../styles/home.css';

function Home() {
  const navigate = useNavigate();
  const { artworks, isLoading } = useSelector((store) => store.artworks);
  const [searchArtwork, setSearchArtwork] = useState('');
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchArtworks());
  }, [dispatch]);

  const filterArtworks = artworks.filter((artwork) => artwork.title
    .toLowerCase()
    .includes(searchArtwork.toLocaleLowerCase()));

  if (isLoading) {
    return <h5>Artworks Loading...</h5>;
  }

  const handleClickedArtwork = (artwork) => {
    navigate(`/details/${artwork.title.toLowerCase()}`, { state: { artwork } });
  };

  return (
    <>
      <div className="flex header">
        <h3 className="title">Art institute of Chicago</h3>
        <form className="flex">
          <input
            className="search"
            type="search"
            value={searchArtwork}
            name="q"
            placeholder="Search Artwork"
            onChange={(e) => setSearchArtwork(e.target.value)}
          />
          <input className="submt-search" type="submit" />
        </form>
      </div>
      <table className="artwork-wrapper">
        <tbody>
          {filterArtworks.map((artwork) => (
            <tr
              className="flex each-artwork"
              key={artwork.id}
              onClick={() => {
                handleClickedArtwork(artwork);
              }}
              onKeyDown={(e) => {
                if (e.key === 'Enter') {
                  handleClickedArtwork(artwork);
                }
              }}
              role="button"
              tabIndex={0}
            >
              <div className="right-arrow">
                <button
                  style={{ color: 'white' }}
                  type="button"
                  aria-label="Go back"
                >
                  <BsArrowRightCircle />
                </button>
              </div>
              <p>{artwork.title}</p>
              <p>{artwork.date_display}</p>
            </tr>
          ))}
        </tbody>
      </table>
    </>
  );
}

export default Home;
