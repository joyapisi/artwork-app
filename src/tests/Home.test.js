import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';
import Home from '../components/Home';
import Store from '../redux/store';
import '@testing-library/jest-dom/extend-expect';

describe('Home', () => {
  test('Render the Home Component', () => {
    const tree = TestRenderer.create(
      <BrowserRouter>
        <Provider store={Store}>
          <Home />
        </Provider>
        ,
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
