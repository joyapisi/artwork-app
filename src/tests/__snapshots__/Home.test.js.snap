// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Render the Home Component 1`] = `
Array [
  <div
    className="flex header"
  >
    <h3
      className="title"
    >
      Art institute of Chicago
    </h3>
    <form
      className="flex"
    >
      <input
        className="search"
        name="q"
        onChange={[Function]}
        placeholder="Search Artwork"
        type="search"
        value=""
      />
      <input
        className="submt-search"
        type="submit"
      />
    </form>
  </div>,
  <div
    className="artwork-wrapper"
  />,
  ",",
]
`;
